service/mantle/shopify/ShopifyOrderServices.xml

```diff
@@ -0,0 +1,285 @@
+            </iterate>
+        </actions>
+    </service>
+
+    <!-- fulfill order -->
+    <service verb="queue" noun="ShopifyFulfillmentSystemMessage" authenticate="anonymous-all">
+        <description>Creates a system message for sending to shopify that decribes orders fulfilled in a shipment</description>
+        <in-parameters>
+            <parameter name="shipmentId" required="true" />
+        </in-parameters>
+        <out-parameters>
+            <parameter name="systemMessageId"/>
+        </out-parameters>
+        <actions>
+            <entity-find-one entity-name="mantle.shipment.Shipment" value-field="shipment" />
+            <if condition="!shipment"><return error="true" message="No shipment found with ID ${shipmentId}"/></if>
+            <if condition="!shipment.statusId in ['ShipShipped','ShipDelivered']">
+                <return error="true" message="ERROR:The shipment ${shipment.shipmentId} must be in status Shipped or Delivered for registering a fulfillment "/>
+            </if>
+
+            <set field="tracking_info" from="[:]"/>
+            <entity-find entity-name="mantle.shipment.ShipmentPackageRouteSeg" list="packageRouteSegList">
+                <econdition field-name="shipmentId"/><order-by field-name="shipmentRouteSegmentSeqId,shipmentPackageSeqId"/></entity-find>
+            <if condition="packageRouteSegList &amp;&amp; packageRouteSegList.first().trackingCode">
+                <set field="tracking_info.number" from="packageRouteSegList.first().trackingCode"/>
+            </if>
+
+            <!-- get the carrierPartyId from the first ShipmentRouteSegment -->
+            <entity-find entity-name="mantle.shipment.ShipmentRouteSegment" list="shipmentRouteSegmentList">
+                <econdition field-name="shipmentId"/><order-by field-name="shipmentRouteSegmentSeqId"/></entity-find>
+            <set field="facilityId" from="shipmentRouteSegmentList.first().originFacilityId"/>
+            <set field="carrierPartyId" from="shipmentRouteSegmentList.first().carrierPartyId"/>
+
+            <entity-find-one entity-name="mantle.party.PartyIdentification" value-field="carrierShopifyId">
+                <field-map field-name="partyId" from="carrierPartyId"/><field-map field-name="partyIdTypeEnumId" value="PtidShopify"/>
+            </entity-find-one>
+            <if condition="carrierShopifyId">
+                <set field="tracking_info.company" from="carrierShopifyId.idValue"/>
+            </if>
+
+            <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="shipmentItemList">
+                <econdition field-name="shipmentId"/></entity-find>
+
+            <entity-find-one entity-name="mantle.facility.Facility" value-field="facility" />
+
+            <!-- get all the shopify fulfillment orders from this shipment -->
+            <set field="orderIdList" from="shipmentItemList.orderId.unique( false )"/>
+            <log message="orderIdList: ${orderIdList}"/>
```

Comment:
May also want to comment this out on merge.

Reply:
the project is still in development, so there will be a few comments in there